pages/api/studentPortal/freemium.ts

```diff
@@ -0,0 +1,25 @@
+import { addDays, differenceInCalendarDays } from 'date-fns';
+
+//used for freemium feature calculations.
+//createdAt comes from fetchUserRole query and trialLength is manually inputted.
+export const trialDaysRemaining = (createdAt:Date, trialLength:number) => {
```

Comment:
rather than putting another file in the studentPortal folder, lets create a subfolder called `freemium` and rename this to index.ts. Or you could break these functions into two different files and name the files after the function names. I prefer this latter approach.

Then you can also create unit tests for both functions. 

These functions are good candidates to create unit tests for. and creating a folder for freemium will isolate all the work you're doing from all the other student portal work.

Reply:
it's always a good idea to test your api functions

Reply:
I created two unit test suites:
trialDaysRemaining.test.ts and elapsedDays.test.ts

Two questions: 
What do I want to return in each function if the created_at date is accidentally modified in the database and it ends up being a date after the current system date?  

I chose to have the trialDaysRemaining function return 0 and the elapsedDays function return 0 --- but wasn't sure if there was a reason to have an undefined so it shows up as NaN??